app/src/test/java/org/mozilla/focus/cfr/CfrMiddlewareTest.kt

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.cfr
+
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.action.RestoreCompleteAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.SecurityInfoState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.support.test.whenever
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+import org.mozilla.focus.Components
+import org.mozilla.focus.state.AppAction
+import org.mozilla.focus.state.AppState
+import org.mozilla.focus.state.AppStore
+import org.mozilla.focus.utils.Settings
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CfrMiddlewareTest {
+    private lateinit var cfrMiddleware: CfrMiddleware
+
+    @Mock
+    private lateinit var middlewareContext: MiddlewareContext<BrowserState, BrowserAction>
+
+    @Mock
+    private lateinit var appStore: AppStore
+
+    @Mock
+    private lateinit var appState: AppState
+
+    @Mock
+    private lateinit var browserStore: BrowserStore
+
+    @Mock
+    private lateinit var browserState: BrowserState
+
+    @Mock
+    private lateinit var components: Components
+
+    @Mock
+    private lateinit var settings: Settings
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        whenever(middlewareContext.store).thenReturn(browserStore)
+        whenever(components.appStore).thenReturn(appStore)
+        whenever(components.appStore.state).thenReturn(appState)
+        whenever(middlewareContext.state).thenReturn(browserState)
+        whenever(components.settings).thenReturn(settings)
+
+        cfrMiddleware = CfrMiddleware(components)
+    }
+
+    @Test
+    fun `GIVEN a new tab is added WHEN erase cfr is enabled and tracking protection cfr is not displayed THEN the numberOfTabsOpened is increased and checked`() {
```

Comment:
Here and below `given` and `when` seems to be missmatched.
- `Given` refers to the initial context, all preconditions
- `When` refers to the event that triggers the outcome we want to test
- `Then` describes the expected outcome that we want to test.

https://cucumber.io/docs/gherkin/reference/
https://cucumber.io/blog/bdd/cucumber-anti-patterns-part-two/#:~:text=No%20clear%20separation%20between%20Given%2C%20When%2C%20and%20Then